### PR TITLE
Update FlexREX for k8s 1.6

### LINKED ISSUE #716: FlexRex adapter script does not display proper error
# Summary
The FlexRex adapter script does not return proper error string when failure occurs in REX-Ray.

# Bug Reports
When an error occurs in REX-Ray, it appears that the error is not being properly reported back to the the FlexVolume plugin to be logged properly in Kubernetes.

## Version
Please paste the output of `rexray version`. For example:

```shell
REX-Ray
-------
Binary: /usr/bin/rexray
Flavor: client+agent+controller
SemVer: 0.7.0
OsArch: Linux-x86_64
Branch: v0.7.0
Commit: a20a838ca70838a914b632637398824fcb10d0db
Formed: Mon, 23 Jan 2017 10:14:32 EST

libStorage
----------
SemVer: 0.4.0
OsArch: Linux-x86_64
Branch: v0.7.0
Commit: a1103d3f215117f7b9f51dae2b24f852c9c54995
Formed: Mon, 23 Jan 2017 10:14:12 EST
```

## Expected Behavior
For instance, when attaching a non-existent volume, the rexray log emits the following error:

```
time="2017-02-08T15:11:39-05:00" level=fatal msg="no volumes found"
```  

## Actual Behavior
The error string displayed in Kubernetes does not reflect what happened in REX-Ray as shown in the following output:
```
MountVolume.SetUp failed for volume "rexray/flexrex/1f871e4d-ee38-11e6-969d-080027859f43-vol-0" (spec.Name: "vol-0") pod "1f871e4d-ee38-11e6-969d-080027859f43" (UID: "1f871e4d-ee38-11e6-969d-080027859f43") with: invalid character '/' looking for beginning of value
```
## Steps To Reproduce
 * Deploy a pod YAML in Kubenertes with a bad volume name similar to the following where volume `xxxtest-vol-1` does not exist.
  ```
apiVersion: v1
kind: Pod
metadata:
  name: pod-0
spec:
  containers:
  - image: gcr.io/google_containers/test-webserver
    name: pod-0
    volumeMounts:
    - mountPath: /test-pd
      name: vol-0
  volumes:
  - name: vol-0
    flexVolume:
      driver: rexray/flexrex
      fsType: ext4
      options:
        volumeID: xxxtest-vol-1
        forceAttach: "true"
        forceAttachDelay: "15"
```
 * Observe the reported error in Kubernetes with `kubectl describe pod`
 * Verify that the REX-Ray log in `/var/log/rexray.log` display the real reason for failure.

### LINKED ISSUE #777: k8s 1.6 Flex Volume support
# Summary
Update FlexREX to handle FlexVolume enhancement in k8s 1.6 (if enhancements are merged)

# New Feature
See: https://github.com/kubernetes/kubernetes/pull/41804

### PR DESCRIPTION
**Pull request by codenrhoden**
This patch adds support for the k8s 1.6 FlexVolume changes. This is by
adding commands for: `waitforattach`, `mountdevice`, `unmountdevice`,
`getvolumename`, and `isattached`.

The debug logging is now handled more centrally.

Fixes #777 

This also Fixes #716, as a more meaningful error is displayed if the volume does not already exist.

### CODE REVIEW

**Comment by codenrhoden** [comment]
This definitely needs some good testing. I am able to use it to successfully run a pod with k8s 1.6, but have not tried on 1.5. The changes are designed with backwards compatibility with 1.5 in mind, so that anyone using FlexREX with 1.5 doesn't break their setup if they upgrade REX-Ray.

File: .docs/user-guide/schedulers.md [hunk 1]
```diff
@@ -159,15 +159,16 @@ configuring popular applications with persistent storage via Docker and REX-Ray.
 REX-Ray can be integrated with [Kubernetes](https://kubernetes.io/) allowing
 pods to consume data stored on volumes that are orchestrated by REX-Ray. Using
 Kubernetes' [FlexVolume](https://kubernetes.io/docs/user-guide/volumes/#flexvolume)
-plug-in, REX-Ray can provide uniform access to storage operatations such as attach,
+plug-in, REX-Ray can provide uniform access to storage operations such as attach,
```
Comment (klausenbusk) [inline .docs/user-guide/schedulers.md]: Shouldn't typo fixes and stuff like that be in a separate commit? It doesn't seems relevant for 1.6 support. :)

**Comment by akutz** [comment]
Hi @codenrhoden,

I would have merged this last week but your comment indicated it needed a lot of testing. I haven't seen this PR updated by you or anyone else to indicate that testing has occurred. Would you please let me know when you're satisfied that this PR is ready to be merged?

**Comment by codenrhoden** [comment]
Your feelings are correct -- I need to hear more from @cduchesne before we do anything with this.

**Comment by akutz** [comment]
Hi @codenrhoden,

Should we retarget this to the release branch or are you thinking it may not make the release and stay pointed at master?

**Comment by codenrhoden** [comment]
retarget. We'll need to get this in.

**Comment by mabounassif** [comment]
How can we set the environment variables that FlexRex depend on (in my case for digital ocean) via kubernetes? Should those environment variables not use Kubernetes secret store, and be set on the barebone level?
